ui/static/ui/js/listing.js

```diff
@@ -1,218 +1,229 @@
-        $('.cd-panel-2').removeClass('is-visible');
-      };
+    var hideTaxonomyPanel = function () {
+      $('.cd-panel-2').removeClass('is-visible');
+    };
 
-      var showTab = function(tabId) {
-        $("a[href='#" + tabId + "'").tab('show');
-      };
+    var showTab = function (tabId) {
+      $("a[href='#" + tabId + "']").tab('show');
+    };
 
-      var setTabName = function(tabId, newTabName) {
-        $("a[href='#" + tabId + "']").find("span").text(newTabName);
-      };
+    var setTabName = function (tabId, newTabName) {
+      $("a[href='#" + tabId + "']").find("span").text(newTabName);
+    };
 
-      var loadManageTaxonomies = function () {
-        ManageTaxonomies.loader(
-          repoSlug,
-          $('#taxonomy-component')[0],
-          showConfirmationDialog,
-          showTab,
-          setTabName,
-          refreshFromAPI
-        );
-      };
+    var showConfirmationDialog = function (options) {
+      var container = $("#confirmation-container")[0];
+      Utils.showConfirmationDialog(
+        options,
+        container
+      );
+    };
 
-      $('[data-toggle=popover]').popover();
-      //Close panels on escape keypress
-      $(document).keyup(function(event) {
-        if (event.keyCode === 27) { // escape key maps to keycode `27`
-          closeLearningResourcePanel();
-          if ($('.cd-panel-2').hasClass('is-visible')) {
-            hideTaxonomyPanel();
-          }
-          if ($('.cd-panel-exports').hasClass('is-visible')) {
-            $('.cd-panel-exports').removeClass('is-visible');
-          }
-          if ($('.cd-panel-members').hasClass('is-visible')) {
-            $('.cd-panel-members').removeClass('is-visible');
+    var ListingContainer = React.createClass({
+      displayName: 'ListingContainer',
+      // make data structure to store query parameters
+      makeQueryMap: function() {
+        var queryMap = {};
+        // Populate queryMap with query string key value pairs.
+        _.each(URI(this.props.getQueryString()).query(true), function (v, k) {
+          if (!Array.isArray(v)) {
+            // URI().query(true) will put two or more values with the same
+            // key in an array and not use an array for single values.
+            // Put everything into arrays for consistency's sake.
+            queryMap[k] = [v];
+          } else {
+            queryMap[k] = v;
           }
-          event.preventDefault();
-        }
-      });
-
-      //close the lateral panel
-      $('.cd-panel').on('click', function (event) {
-        if ($(event.target).is('.cd-panel') ||
-          $(event.target).is('.cd-panel-close')) {
-          closeLearningResourcePanel();
-          event.preventDefault();
+        });
+        return queryMap;
+      },
+      // Return query portion of URL
+      makeQueryString: function(queryMap) {
+        var newQuery = "?" + URI().search(queryMap).query();
+        if (newQuery === "?") {
```

Comment:
I changed it back since it looks like I'm setting some values to `undefined` which don't show up in the query string but would make `queryMap` non-empty. Hooray for unit tests!